File: 2 SEMESTRE/TADS/FILA/FilaEstatica.c
/*Código para mostrar como funciona uma fila estática, basicamente a FILA consiste em adicionar um elemento no fim e remover do início
FIFO = First In First Out, como se imaginassimos uma fala onde o primeiro que entra é o primeiro que sai, e as pessoas vão chegando atras, 
por isso o inserir é no início.
*/

#include <stdio.h> 
#include <stdlib.h> 

//Definir tamanho da lista
#define MAX 7

//Struct para armazenar o array[] e a o tamanho da fila
typedef struct { 
    int array[MAX]; //vetor que guarda a capacidade max da lista
    int n; //quantidade de elementos da lista
} Fila; 

//incializar fila
void inicializa (Fila *fila) {
    //Inicializar quantidade de elementos como 0
    fila->n = 0; 
}

//Procedimento para inserir no fim da fila
void inserir (Fila *fila, int elemento) {
    //Verificar se a fila está cheia
    if (fila->n < MAX) {
        //Adicionar elemento no fim da fila 
        fila->array[fila->n] = elemento; 
        fila->n++; 
    } else {
        printf("Full List"); 
    }
}

//Procedimento para remover do início da lista. 
void remover (Fila *fila) {
    //Variável para armazenar o elemento removido da fila
    int elementoRem; 

    //Verificar se a fila está vazia
    if (fila->n > 0) {
        //Elemento removido vai ser o elemento da pos inicial. 
        elementoRem = fila->array[0]; 

        //Deslocar os elementos para esquerda
        for (int i = 0; i < fila->n; i++) {
            fila->array[i] = fila->array[i + 1]; 
        }

        //Decrementar o contador, para diminuir a quantidade de elementos
        fila->n--; 
    } else {
        printf("Void List"); 
    }

    return elementoRem; 
}

//Main
int main() {
    Fila fila; 

    //Chamar método para incializar fila
    inicializa (&fila); 

    //Métodos para inserir e remover... 
}

File: 2 SEMESTRE/TADS/FILA/FilaEstatica.java
/*Mesmo exercício porém na linguagem de java */

import java.util.Scanner; 

public class FilaEstatica {
    //Atributos da fila
    private int array[]; 
    private int n; 

    //Método para inicializar o tamano da fila
    public static void inicializa(int tamanho) {
        //Inicializar lista como vazia
        this.array = new int (tamanho); 
        this.n = 0; 
    }

    //Método para inserir no fiim da fila
    public static void inserir(int x) throws Exception {
        //Verificar se a fila está cheia
        if (n >= array.length) {
            throw new Exception ("Full List"); 
        }

        //Inserir no fim da fila
        array[n] = x; 
        n++; 
    }

    //Método para remover do inicio da fila
    public static void remover() throws Exception { 
        //Verificar se a fila está vazia. 
        if (n == 0) {
            throw new Exception ("Void List"); 
        }

        //Elemento removido vai ser o elemento inicial
        int elementoRem = array[0]; 

        //Deslocar os elementos para a esquerda
        for (int i = 0; i < n - 1; i++) {
            array[i] = array[i + 1]; 
        }

        //Decrementar o contador
        n--; 

        //Retornar o elemento removido
        return elementoRem; 
    }

    // Método auxiliar para ajudar você a testar e ver a fila
    public void mostrar() {
       //Mostrar a fila preenchida
        for (int i = 0; i < n; i++) {
            System.out.print(array[i] + " ");
        }
        System.out.println();
    }
}